src/rpc/utils.rs

```diff
@@ -0,0 +1,45 @@
+use crate::rpc::result::internal_rpc_err;
+use jsonrpsee::core::RpcResult;
+
+pub fn compare_values<T: std::cmp::PartialEq + std::fmt::Display>(
+    name: &str,
+    expected: T,
+    actual: T,
+) -> RpcResult<()> {
+    if expected != actual {
+        Err(internal_rpc_err(format!(
+            "incorrect {} {}, expected {}",
+            name, actual, expected
+        )))
+    } else {
+        Ok(())
+    }
+}
+
+// TODO: Can we avoid hard coding this here and read it from reth config instead ?
+const GAS_LIMIT_BOUND_DIVISOR: u64 = 1024;
+const MIN_GAS_LIMIT: u64 = 5000;
+
+// Compute the gas limit of the next block after parent. It aims
+// to keep the baseline gas close to the provided target, and increase it towards
+// the target if the baseline gas is lower.
+// Reference: https://github.com/flashbots/builder/blob/03ee71cf0a344397204f65ff6d3a917ee8e06724/core/utils/gas_limit.go#L8
+pub fn calc_gas_limit(parent_gas_limit: u64, desired_limit: u64) -> u64 {
+    let delta = parent_gas_limit / GAS_LIMIT_BOUND_DIVISOR - 1;
+    let mut limit = parent_gas_limit;
+    let desired_limit = std::cmp::max(desired_limit, MIN_GAS_LIMIT);
+    if limit < desired_limit {
```

Comment:
This code is confusing 😅 (and yes, I know you just copied it, makes total sense to me, but let's see if we can improve it a little). This should simply say `parent_gas_limit < desired_limit`. Same in the next if statement. If we then also early return in the next statement we can make the final statement in this fn simply `parent_gas_limit` too, and now we can drop the mutated `limit` entirely.

I asked CGPT, it agrees, it simplifies the code to:
```rust
let delta = parent_gas_limit / GAS_LIMIT_BOUND_DIVISOR - 1;
let desired_limit = std::cmp::max(desired_limit, MIN_GAS_LIMIT);

if parent_gas_limit < desired_limit {
    std::cmp::min(parent_gas_limit + delta, desired_limit)
} else if parent_gas_limit > desired_limit {
    std::cmp::max(parent_gas_limit - delta, desired_limit)
} else {
    parent_gas_limit
}
```

Personally, I'd go one step further, I think this is the most clear and compressed version:
```rust
let delta = parent_gas_limit / GAS_LIMIT_BOUND_DIVISOR - 1;
let desired_limit = std::cmp::max(desired_limit, MIN_GAS_LIMIT);

if parent_gas_limit < desired_limit {
    let max_limit = parent_gas_limit + delta;
    std::cmp::min(max_limit, desired_limit)
} else if parent_gas_limit > desired_limit {
    let min_limit = parent_gas_limit - delta;
    std::cmp::max(min_limit, desired_limit)
} else {
    parent_gas_limit
}
```

Reply:
Let me know if the test passes 😅 .

Reply:
Naming could be a little better still. `min_acceptable_limit` and `max_acceptable_limit` to indicate this is unrelated to the chain's min_gas_limit of 5_000.

Reply:
Good point with using the min / max functions instead of the confusing implementation with an if block.

Reply:
Implemented in: https://github.com/ultrasoundmoney/reth-payload-validator/pull/18/commits/54fc29893313e32e8b70ebf89f3bb4f198815290

Reply:
Clippy made me use a match statement instead: https://github.com/ultrasoundmoney/reth-payload-validator/pull/18/commits/c4390927fcfefa1330acb87b08d1c36f30d238ac